### PR TITLE
Kamu 150 add set info event to dataset from UI

### PR DESCRIPTION
**Pull request by dmitriy-borzenko**

### CODE REVIEW

File: src/app/dataset-view/additional-components/overview-component/overview-component.html [hunk 1]
```diff
@@ -6,32 +6,154 @@
...
+                        aria-label="Edit dataset details"
+                        data-test-id="edit-dataset-details"
+                        title="Edit dataset details"
+                        (click)="openInformationModal(content)"
+                    >
+                        <svg-icon name="pencil"></svg-icon>
+                    </button>
+                </div>
+            </div>
+            <ng-template #content let-modal>
```
Comment (zaychenko-sergei) [inline src/app/dataset-view/additional-components/overview-component/overview-component.html:59]: Please move the modal window to a separate component

File: src/app/dataset-view/additional-components/overview-component/overview-component.ts [hunk 2]
```diff
@@ -84,4 +117,48 @@ export class OverviewComponent extends BaseComponent implements OnInit {
...
+            this.yamlSetInfoTemplate += `  description: ${this.description}\n`;
+        if (this.keywords.length) {
+            this.yamlSetInfoTemplate += `  keywords:\n`;
+            this.keywords.forEach(
+                (keyword: string) =>
+                    (this.yamlSetInfoTemplate += `    - ${keyword}\n`),
+            );
+        }
+        return this.yamlSetInfoTemplate;
+    }
```
Comment (zaychenko-sergei) [inline src/app/dataset-view/additional-components/overview-component/overview-component.ts:152]: This does not belong to visual components. Maybe a separate utility service.

**Review by zaychenko-sergei — CHANGES_REQUESTED** [review]
Tests + please fix test regressions

**Comment by zaychenko-sergei** [comment]
If I add a description once, I need to have a possibility to clear it.

File: src/app/dataset-view/additional-components/overview-component/overview-component.html (hunk 1 above)
Comment (dmitriy-borzenko) [reply src/app/dataset-view/additional-components/overview-component/overview-component.html:59]: Moved  the modal window to a separate component

File: src/app/dataset-view/additional-components/overview-component/overview-component.ts (hunk 2 above)
Comment (dmitriy-borzenko) [reply src/app/dataset-view/additional-components/overview-component/overview-component.ts:152]: Moved this method to separate service

File: src/app/dataset-view/additional-components/overview-component/components/details-modal/details-modal.component.html [hunk 3]
```diff
@@ -58,6 +58,7 @@
         class="btn btn-success"
         (click)="commitSetInfoEvent(); activeModal.close()"
         data-test-id="commit-event"
+        [disabled]="isDetailsNotChanged"
```
Comment (zaychenko-sergei) [inline src/app/dataset-view/additional-components/overview-component/components/details-modal/details-modal.component.html:61]: `unmodifiedDetails`

**Review by zaychenko-sergei — APPROVED** [review]

**Pull request merged**